include/quicly.h

```diff
@@ -969,6 +981,12 @@ static int quicly_stream_has_receive_side(int is_client, quicly_stream_id_t stre
  *
  */
 static int quicly_stream_is_self_initiated(quicly_stream_t *stream);
+/**
+ * Registers a datagram frame payload to be sent. When the applications calls `quicly_send` the first time after registering the
+ * datagram frame payload, the payload is either sent or the reference is discarded. Until then, it is the caller's responsibility
+ * to retain the memory pointed to by `payload`. At the moment, DATAFRAM frames are not congestion controlled.
+ */
+void quicly_set_datagram_frame(quicly_conn_t *conn, ptls_iovec_t payload);
```

Comment:
The assumption behind the design is that applications using datagram frames would want to send data immediately. This is in contrast to stream data that is assumed to be buffered and prioritized.

Applications would build the payload, call `quicly_set_datagram_frame`, then `quicly_send` immediately afterwards.